src/control/server/external.go

```diff
@@ -56,6 +59,7 @@ type External interface {
 	writeToFile(string, string) error
 	createEmpty(string, int64) error
 	mount(string, string, string, uintptr, string) error
+	isMountPoint(string) (bool, error)
```

Comment:
I have serious misgivings about this ever-expanding interface. This isn't really the intent for how interfaces were designed to be used in Go. We're reimplementing core stdlib functionality in order to inject mock implementations for testing, but this is increasingly unwieldy and not sustainable. It's introducing tight coupling between code which should have distinct and unrelated areas of responsibility. All that having been said -- I'm not going to hold up this patch due to those concerns. Solving that problem is going to require a larger discussion and effort to get us set up along a different path.

Reply:
agreed, let's look at alternative ways of doing this.